src/comm/ConnOpener.cc

```diff
@@ -431,7 +432,7 @@ Comm::ConnOpener::lookupLocalAddress()
 {
     struct sockaddr_storage addr = {};
     socklen_t len = sizeof(addr);
-    if (getsockname(conn_->fd, reinterpret_cast<struct sockaddr *>(&addr), &len) != 0) {
+    if (xgetsockname(conn_->fd, reinterpret_cast<struct sockaddr *>(&addr), &len) != 0) {
```

Comment:
How do you feel about providing a second xgetsockname() call that accepts a sockaddr_storage reference instead of the standard two parameters? That trivial wrapper will remove many reinterpret_casts that this PR currently has to leave behind...

```suggestion
    if (xgetsockname(conn_->fd, addr) != 0) {
```

Or is it best to leave that as a followup because it would require removing a few no-longer needed address length variables and memset() calls from xgetsockname() callers as well?

Reply:
I'd do as a followup, there's scope creep enough here